Box_intersection_d/include/CGAL/Box_intersection_d/Box_d.h

```diff
@@ -36,14 +38,16 @@ namespace Box_intersection_d {
 template <class Dummy>
 struct Unique_numbers {
     typedef std::size_t ID;
-    Unique_numbers() : i(n++) {}
+    Unique_numbers() {
+      static CGAL::cpp11::atomic<std::size_t> n; // initialized to 0
+      i = n++;
+    }
```

Comment:
It means that those "unique IDs" will really be unique, even if there are multiple threads.